rust/src/client.rs

```diff
@@ -156,6 +156,47 @@ pub(crate) fn client_handle_fd_argument(
     }
 }
 
+/// Connect to the client socket and ensure the daemon is initialized;
+/// this avoids DBus and ensures that we get any early startup errors
+/// returned cleanly.
+#[cfg(feature = "client-socket")]
+fn start_daemon_via_socket() -> Result<()> {
+    use cap_std::io_lifetimes::IntoSocketlike;
+
+    let conn = tokio::net::UnixStream::connect("/run/rpm-ostree/client.sock")?;
```

Comment:
`tokio::net::UnixStream::connect("/run/rpm-ostree/client.sock").await?` will turn this function into an async

Reply:
Right, we can probably just make it synchronous and use `std` here.

Reply:
Trying to `connect` after the daemon is idle or closed results in a `os error 111: connection refused`. This is because the daemon is no longer listening on the socket